Add documentation for changes introduced by jslib-aws v0.7.0 (#939)
grafana/k6-docs: src/data/markdown/docs/20 jslib/01 jslib/01 aws/06 SignatureV4.md

---
title: 'SignatureV4'
head_title: 'SignatureV4'
description: 'SignatureV4 is used to sign or pre-sign requests to AWS services using the Signature V4 algorithm'
excerpt: 'SignatureV4 is used to sign and pre-sign requests to AWS services using the Signature V4 algorithm'
---

> SignatureV4 allows producing authenticated HTTP requests to AWS services. Namely, it allows to sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request, with authorization information stored in its headers, while the `presign` operation produces a pre-signed request, with authorization information stored in its query string parameters.

Reviewer: ```suggestion
With SignatureV4, you can  produce authenticated HTTP requests to AWS services. Namely, it lets you to sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request, with authorization information stored in its headers, while the `presign` operation produces a pre-signed request, with authorization information stored in its query string parameters.
```

> SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the services clients.

Reviewer: ```suggestion
SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the service's clients.
```

| applyChecksum | boolean                                                                                   | Whether to calculate a checksum of the request body and include it as either a request header (when signing) or as a query string parameter (when pre-signing). This is required for AWS Glacier and Amazon S3 and optional for every other AWS service as of late 2017. |

## Methods

| Method                                                     | Description                                                                   |
| :--------------------------------------------------------- | :---------------------------------------------------------------------------- |
| [sign()](/javascript-api/jslib/aws/signaturev4/sign)       | Signs an authenticated HTTP request using the AWS signature v4 algorithm      |
| [presign()](/javascript-api/jslib/aws/signaturev4/presign) | Produces an authenticated pre-signed URL using the AWS signature v4 algorithm |

### Throws

> SignatureV4 methods will throw errors in case of failure.

Reviewer: ```suggestion
SignatureV4 methods throw errors on failure.
```

---
title: 'SignatureV4'
head_title: 'SignatureV4'
description: 'SignatureV4 is used to sign or pre-sign requests to AWS services using the Signature V4 algorithm'
excerpt: 'SignatureV4 is used to sign and pre-sign requests to AWS services using the Signature V4 algorithm'
---

> With SignatureV4, you can  produce authenticated HTTP requests to AWS services. Namely, it lets you to sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request, with authorization information stored in its headers, while the `presign` operation produces a pre-signed request, with authorization information stored in its query string parameters.
> 
> SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the service's clients.
> 
> Instantiating a new `SignatureV4` requires a single options object argument with the following properties: 

Reviewer: ```suggestion
With SignatureV4, you can produce authenticated HTTP requests to AWS services. Namely, it lets you sign and pre-sign requests to AWS services using the [Signature V4](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) algorithm. The `sign` operation produces a signed request with authorization information stored in its headers.
The `presign` operation produces a pre-signed request with authorization information stored in its query string parameters.

SignatureV4 is included in both the dedicated jslib `signature.js` bundle and the `aws.js` one, containing all the service's clients.

Instantiating a new `SignatureV4` requires a single options object argument with the following properties: 
```

| service       | string                                                                                    | the AWS region to sign or pre-sign requests for. As described by [Amazon AWS docs](https://docs.aws.amazon.com/general/latest/gr/rande.html)                                                                                                                             |
| region        | string                                                                                    | the AWS service to sign or pre-sign requests for. As described by [Amazon AWS docs](https://aws.amazon.com/about-aws/global-infrastructure/regions_az/)]                                                                                                                  |
| credentials   | an object with `accessKeyId`, `secretAccessKeyId`, and optional `sessionToken` properties | the AWS credentials to sign or pre-sign requests with.                                                                                                                                                                                                                   |

> | uriEscapePath | boolean                                                                                   | Whether to uri-escape the request URI path as part of computing the canonical request string. This is required for every AWS service, except Amazon S3, as of late 2017.                                                                                                |

Reviewer: ```suggestion
| uriEscapePath | boolean                                                                                   | Whether to uri-escape the request URI path as part of computing the canonical request string. As of late 2017, this is **required for every AWS service** except Amazon S3.                                                                                                |
```

> | applyChecksum | boolean                                                                                   | Whether to calculate a checksum of the request body and include it as either a request header (when signing) or as a query string parameter (when pre-signing). This is required for AWS Glacier and Amazon S3 and optional for every other AWS service as of late 2017. |

Reviewer: ```suggestion
| applyChecksum | boolean                                                                                   | Whether to calculate a checksum of the request body and include it as either a request header (when signing) or as a query string parameter (when pre-signing). This is **required for AWS Glacier and Amazon S3** and optional for every other AWS service as of late 2017. |
```
